speak/face.py

```diff
@@ -152,20 +144,37 @@ def update(self, status=None):
             self._eyebox.pack_start(the, True, True, FACE_PAD)
             the.show()
 
-        self._mouth = status.mouth(self._audio, self.fill_color)
+        self._mouth = status.mouth(self.fill_color)
         self._mouth.show()
         self._mouthbox.add(self._mouth)
 
     def say(self, something):
-        self._audio.speak(self._peding or self.status, something)
+        if self._pending is None:
+            voice_name = self.status.voice.name
+        else:
+            voice_name = self._pending.voice.name
+        all_voices = self.speech.get_all_voices()
+        lang_code = None
+        for lang, name in all_voices.items():
+            if name == voice_name:
+                lang_code = lang
+        self.speech.say_text(something, pitch=None,
+                             rate=None, lang_code=lang_code)
 
     def say_notification(self, something):
-        status = (self._peding or self.status).clone()
+        status = (self._pending or self.status).clone()
```

Comment:
A clone of a Status object is created, then discarded by the end of this function.  I don't see why the attributes of the original Status object are obtained instead of using a clone.  Any idea why?

Reply:
@quozl 
I'm not sure why.
`say_notification` is used to select the language from the combobox in the game settings, whereas `say` is used when the card flip is made.

In the `say_notification` function, the `voice` attribute of the status object is being reassigned. Perhaps the idea was to prevent sideeffects being introduced by using the same `Status` object for both the above mentioned tasks.

Reply:
I'll work backwards and explain what I see as of e71e3be42dec84409af16f8b3318809598aeb19e.

`say_text` is to be given something to say, and a language code for the voice to use to say it.  Rather than store the language code, the activity stores the voice name.  Converting the voice name to the language code uses a `for` loop.

`say` is to be given something to say, and it will use the currently chosen voice name for the card; which it obtains from an instance of `Status`.

`say_notification` is to be given something to say, and it will use the `defaultVoice()`, which it obtains from a cloned instance of `Status`.

Just speculating here, but perhaps `say_notification` could call `speech.say_text` without a language code, causing the default voice or system language to be used.